### PR TITLE
AddingPrecautionsEndDate

### PR DESCRIPTION
**Pull request by AlexKaschuk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs [hunk 1]
```diff
@@ -107,5 +113,18 @@ public async Task CheckIfAdminAsync(User user)
                 throw new UnauthorizedAccessException();
         }
 
+        private async Task<IEnumerable<UserPrecaution>> CheckEndDate(IEnumerable<UserPrecaution> userPrecaution) {
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs:116]:
```suggestion
        private async Task<IEnumerable<UserPrecaution>> CheckEndDateAsync(IEnumerable<UserPrecaution> userPrecaution) {
```

File: EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs [hunk 2]
```diff
@@ -107,5 +113,18 @@ public async Task CheckIfAdminAsync(User user)
...
+            foreach (var item in userPrecaution)
+            {
+                if (item.EndDate < DateTime.Now&& item.IsActive != false)
```
Comment (voroniak) [inline EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs:119]:
```suggestion
                if (item.EndDate < DateTime.Now && item.IsActive)
```

**Review by voroniak — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs (hunk 1 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs]: done

File: EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs (hunk 2 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.BLL/Services/Precautions/UserPrecautionService.cs]: done

**Review by voroniak — APPROVED** [review]

**Review by randomVovaKuhot — APPROVED** [review]

**Review by sooooolo — APPROVED** [review]

**Pull request merged**